lib/asynchronize.rb

```diff
@@ -11,13 +11,13 @@ def self.included(base)
       #   - The new methods wrap the old method within Thread.new.
       #   - Subsequent calls only add methods to the existing Module.
       #
-      # @param methods [Symbol] The methods to be asynchronized.
-      # @example To add any number of methods to be asynchronized.
+      #   @param methods [Symbol] The methods to be asynchronized.
+      #   @example To add any number of methods to be asynchronized.
       #   asynchronize :method1, :method2, :methodn
       #
       def self.asynchronize(*methods)
         # require 'pry'; binding.pry
-        module_name = self.name.split('::').last + 'Asynchronized'
+        module_name = Asynchronize.get_container_name(self.name)
```

Comment:
I think I like the version we discussed better.. `async_container = Asynchronize.get_container_for(self)`

Where that would be responsible for  getting or defining and prepending the module.